core/modules/replica/IngestHttpSvcMod.cc

```diff
@@ -155,19 +167,91 @@ json IngestHttpSvcMod::_readLocal(string const& filename) {
 }
 
 
-json IngestHttpSvcMod::_readRemote(string const& method,
+json IngestHttpSvcMod::_readRemote(string const& database,
+                                   string const& method,
                                    string const& url,
                                    string const& data,
                                    vector<string> const& headers) {
     debug(__func__);
     size_t numBytes = 0;
     size_t numRows = 0;
-    HttpFileReader reader(method, url, data, headers);
-    reader.read([this,&numBytes,&numRows](string const& row) {
+
+    // Check if SSL peer certificate management was requested for the catalog.
+    // Ignore missing parameters if they were not set.
+    auto const databaseServices = serviceProvider()->databaseServices();
+    bool sslVerifyPeer = true;
+    try {
+        sslVerifyPeer = stoi(databaseServices->ingestParam(
+                database,
+                HttpFileReaderConfig::category,
+                HttpFileReaderConfig::sslVerifyPeerKey).value) != 0;
+    } catch (DatabaseServicesNotFound const&) {}
+
+    string caPath;
+    string caInfo;
+    string caInfoTmp;
+    if (sslVerifyPeer) {
+        // Get a path to a folder with host certificates, if any was configured
+        // for this catalog.
+        try {
+            caPath = databaseServices->ingestParam(
+                    database,
+                    HttpFileReaderConfig::category,
+                    HttpFileReaderConfig::caPathKey).value;
+        } catch (DatabaseServicesNotFound const&) {}
+
+        // Get a path to an existing file with peer certificate, if any was configured
+        // for this catalog.
+        try {
+            caInfo = databaseServices->ingestParam(
+                    database,
+                    HttpFileReaderConfig::category,
+                    HttpFileReaderConfig::caInfoKey).value;
```

Comment:
This is ignored if `caInfoValKey` exists, maybe reverse the order and not try to retrieve it if `caInfoValKey` is there, will save one trip to database?